src/openforms/forms/migrations/0042_explanation_template_data_migration.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.2.14 on 2022-08-26 15:29
+from django.db import migrations
+
+
+def append_explanation_template_message(apps, schema_editor):
+    from openforms.authentication.registry import register as auth_register
+    from django.utils.translation import gettext_lazy as _
+
+    Form = apps.get_model("forms", "Form")
+    for form in Form.objects.all():
+        plugins = form.authentication_backends
+        login_options_count = len(
+            [plugin_id for plugin_id in plugins if plugin_id in auth_register._registry]
+        )
+        can_login = login_options_count > 0
+        form_login_required = any(
+            form_step.form_definition.login_required
+            for form_step in form.formstep_set.all()
+        )
+
+        if form_login_required:
+            message = _("Please authenticate to start the form.")
+        elif can_login:
+            message = _("Please authenticate or start the form anonymously.")
+        else:
+            message = _("Please click the button below to start the form.")
+
+        form.explanation_template += f'<p id="start-form">{message}</p>'
+        form.save()
+
+
+def remove_explanation_template_message(apps, schema_editor):
+    from bs4 import BeautifulSoup
+
+    Form = apps.get_model("forms", "Form")
+
+    for form in Form.objects.all():
+        soup = BeautifulSoup(form.explanation_template, "html.parser")
```

Comment:
we can use lxml, it's a dependency of the project anyway.

Reply:
We can, but we don't need the extra functionality offered by lxml.

Reply:
Except that lxml uses C-bindings and is much more performant in parsing the HTML, that alone is enough reason to use it if you have it.

Reply:
lxml will add html and body tags to the string. This is a migration file and we are not concerned about a small gain in performance.

Reply:
> lxml will add html and body tags to the string

now _that's_ an actual valid reason to consider not using it, because it's too much hassle to strip those out.

> This is a migration file and we are not concerned about a small gain in performance.

Yes we are, for those environments were downtime is not acceptable.

The application runs migrations during startup. While the app is migrating, it is NOT available as a backend to serve traffic. Migrations should complete as fast as possible.

Reply:
I'm going to be practical here, and we can remove the backward migration entirely. This because the `id` will also not be used (see above).

NOTE: This is a PO-decision not a technical decision :)